collector/mdadm_linux.go

```diff
@@ -158,19 +175,28 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 		}
 		currentMD = mainLine[0]               // The name of the md-device.
 		isActive := (mainLine[2] == "active") // The activity status of the md-device.
-		personality = mainLine[3]             // The personality type of the md-device.
+		personality = ""
+		for _, possiblePersonality := range mainLine {
+			if raidPersonalityRE.MatchString(possiblePersonality) {
+				personality = possiblePersonality
+				// break
```

Comment:
I don't think a raid array can have many personalities. Today the code isn't really setup to handle that either, it determines the personality of array and then uses one of the other REs to locate the data it wants. I think this break got juggled around with the merge conflicts, I'm going to add it back in.